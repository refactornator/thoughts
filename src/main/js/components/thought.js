import React from 'react';
import moment from 'moment';
import ActionDelete from 'material-ui/svg-icons/action/delete';
import IconButton from 'material-ui/IconButton';
import {Card, CardHeader, CardText} from 'material-ui/Card';

export default class Thought extends React.Component {
    render() {
        const style = Object.assign({}, {marginTop: 10});
        const sinceHappened = moment(this.props.thought.creationTime).fromNow();
        return (
            <Card style={style}>
                <CardHeader subtitle={sinceHappened} style={{height: 20}}>
                    <IconButton className="delete-button" iconStyle={{color: 'rgba(0, 0, 0, 0.541176)', width: 20, height: 20}}>
                        <ActionDelete onClick={this.props.deleteThoughtHandler} />
                    </IconButton>
                </CardHeader>
                <CardText>
                    {this.props.thought.text}
                </CardText>
            </Card>
        )
    }
}