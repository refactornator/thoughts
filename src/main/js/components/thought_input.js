import React from 'react';
import { connect } from 'react-redux';
import Toggle from 'material-ui/Toggle';
import { Card, CardText } from 'material-ui/Card';
import RaisedButton from 'material-ui/RaisedButton';
import LocalStorage from '../utilities/local_storage';

import { createThought, changeThought, changeCategory } from '../actions';

const styles = {
  label: {
    float: 'left',
    marginLeft: 30,
    marginRight: 10,
    marginTop: 8,
    color: 'rgba(0, 0, 0, 0.870588)'
  },
  toggle: {
    marginTop: 7,
    width: null,
    display: 'inline-block',
    marginBottom: 16
  }
};

class ThoughtInput extends React.Component {
  handleNewThought() {
    this.props.createThought({
      category: this.props.category,
      text: this.props.thought
    });
  }

  handleToggleChange(event, toggled) {
    const newCategory = toggled ? 'life' : 'work';
    this.props.changeCategory(newCategory);
  }

  render() {
    const { category, thought, changeThought } = this.props;
    return (
      <Card className="thought-input" style={{ zIndex: 100 }}>
        <CardText style={{ padding: '9px 16px 0 16px' }}>
          <textarea
            value={thought}
            onChange={event => changeThought(event.target.value)}
          />
          <label style={styles.label}>Work</label>
          <Toggle
            label="Life"
            toggled={category === 'work' ? false : true}
            labelPosition="right"
            onToggle={this.handleToggleChange.bind(this)}
            style={styles.toggle}
          />
          <div style={{ textAlign: 'right', float: 'right' }}>
            <RaisedButton
              className="remember-button"
              primary={true}
              label="Remember"
              onClick={this.handleNewThought.bind(this)}
            />
          </div>
        </CardText>
      </Card>
    );
  }
}

export default connect(({ category, thought }) => ({ category, thought }), {
  createThought,
  changeThought,
  changeCategory
})(ThoughtInput);
