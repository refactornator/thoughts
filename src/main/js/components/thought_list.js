import React from 'react';

import Thought from './thought';

export default class ThoughtList extends React.Component {
    render() {
        return (
            <ul className="thoughts">
                {this.props.thoughts.map(({key, style, data}) => {
                    let deleteThoughtHandler = this.props.deleteThoughtHandler.bind(this, data._links.self.href);
                    return <Thought
                        key={key}
                        style={style}
                        thought={data}
                        deleteThoughtHandler={deleteThoughtHandler}
                    />;
                })}
            </ul>
        )
    }
}
