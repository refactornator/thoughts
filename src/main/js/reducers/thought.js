import * as ActionTypes from '../ActionTypes';

import LocalStorage from '../utilities/local_storage';

export default function category(
  state = LocalStorage.getItem('thought') || '',
  action
) {
  switch (action.type) {
    case ActionTypes.CHANGE_THOUGHT:
      LocalStorage.setItem('thought', action.payload);
      return action.payload;
    case ActionTypes.SUCCESSFULLY_CREATED_THOUGHT:
      LocalStorage.removeItem('thought');
      return '';
    default:
      return state;
  }
}
