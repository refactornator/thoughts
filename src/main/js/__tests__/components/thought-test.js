import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ActionDelete from 'material-ui/svg-icons/action/delete';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import { Thought } from '../../components/thought';

import injectTapEventPlugin from 'react-tap-event-plugin';
injectTapEventPlugin();

describe('Thought', () => {
  const thought = {
    id: 1,
    text: 'This is a thought.',
    creationTime: new Date().toISOString()
  };

  it('renders a single thought', () => {
    const renderedThought = renderer
      .create(
        <MuiThemeProvider>
          <Thought thought={thought} forgetThought={() => {}} />
        </MuiThemeProvider>
      )
      .toJSON();

    expect(renderedThought).toMatchSnapshot();
  });

  describe('when the delete button is pressed', () => {
    it('calls the forgetThought prop callback', () => {
      const forgetThoughtMock = jest.fn();
      const renderedThought = shallow(
        <Thought thought={thought} forgetThought={forgetThoughtMock} />
      );

      renderedThought.find(ActionDelete).simulate('click');

      expect(forgetThoughtMock).toBeCalledWith(1);
    });
  });
});
