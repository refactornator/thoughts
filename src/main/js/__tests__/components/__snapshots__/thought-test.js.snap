// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thought renders a single thought 1`] = `
<li
  className="thought-card"
>
  <h6>
    <div
      className="relative-time"
    >
      a few seconds ago
    </div>
    <button
      className="delete-button"
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "rgba(0, 0, 0, 0.541176)",
              "display": "inline-block",
              "fill": "currentColor",
              "height": 20,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 20,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
          />
        </svg>
      </div>
    </button>
  </h6>
  <p>
    This is a thought.
  </p>
</li>
`;
